torch/utils/data/sampler.py

```diff
@@ -90,6 +123,39 @@ def __len__(self):
         return self.num_samples
 
 
+class ClassAwareSampler(Sampler):
+    """Samples elements randomly, without replacement.
+
+    Arguments:
+        data_source (Dataset): dataset to sample from
+
+    Implemented Class-Aware Sampling: https://arxiv.org/abs/1512.05830
+    Li Shen, Zhouchen Lin, Qingming Huang, Relay Backpropagation for Effective
+    Learning of Deep Convolutional Neural Networks, ECCV 2016
+    By default num_samples equals to number of samples in the largest class
+    multiplied by num of classes such that all samples can be sampled.
+    """
+
+    def __init__(self, data_source, num_samples=0):
+        self.data_source = data_source
+        n_cls = len(data_source.classes)
```

Comment:
This makes the assumption that `data_source` has a `classes` field, which is not in general valid